x-pack/plugins/security_solution/server/lib/detection_engine/signals/search_after_bulk_create.ts

```diff
@@ -235,13 +235,9 @@ export const searchAfterAndBulkCreate = async ({
             buildRuleMessage(`filteredEvents.hits.hits: ${filteredEvents.hits.hits.length}`)
           );
 
-          if (
-            filteredEvents.hits.hits[0].sort != null &&
-            filteredEvents.hits.hits[0].sort.length !== 0
-          ) {
-            sortId = filteredEvents.hits.hits[0].sort
-              ? filteredEvents.hits.hits[0].sort[0]
-              : undefined;
+          const lastSortId = filteredEvents.hits.hits[filteredEvents.hits.hits.length - 1].sort;
```

Comment:
While we're changing this code, I think we should always use the unfiltered `searchResult` below to get the `searchAfter` value. If we use `filteredEvents` for `searchAfter`, then on the next search we may get back some useless results that were already filtered out in this iteration.

So I think we can get rid of this block entirely and move the sortId calculations outside of the `else` block below.

Reply:
Totally agree @marshallmain, I went ahead and changed that and a few other tech debt items in this file so its healthier now.